src/app/frontend/replicasetlist/replicasetlist_state.js

```diff
@@ -26,6 +26,22 @@ export default function stateConfig($stateProvider) {
     controller: ReplicaSetListController,
     controllerAs: 'ctrl',
     url: '/replicasetlist',
+    resolve: {
+      replicaSets: resolveReplicaSets,
+    },
     templateUrl: 'replicasetlist/replicasetlist.html',
   });
+
+  /**
+   * Resolves replica sets for the deploy view.
+   *
+   * @param {!angular.$resource} $resource
+   * @ngInject
+   */
+  function resolveReplicaSets($resource) {
```

Comment:
The only thing I'm not sure about is whether this should be inner function or should be on the file scope. I'd personally move this to file scope. We're using ES6 modules, so there is no possibility for name collision and such pattern would allow us to test this function (when needed).

What do you think? If you agree, can you refactor namespace resolving function to follow the same pattern?